File: src/Systems/Fortran/patterns/object_pool.f90
module object_pool_example
  implicit none
contains
  logical function run()
    integer :: pool(2), borrowed
    pool = [1,2]
    borrowed = pool(2); pool(2) = 0; pool(2) = borrowed
    run = all(pool == [1,2]) .and. borrowed == 2
  end function
end module

File: src/Enterprise/VB.NET/patterns/ObjectPool.vb
Imports System.Collections.Generic
Friend Module ObjectPoolExample
    Friend Function Run() As Boolean
        Dim pool As New Stack(Of Integer)(New Integer(){1,2})
        Dim x=pool.Pop()
        pool.Push(x)
        Return pool.Count=2 AndAlso pool.Contains(x)
    End Function
End Module
